components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -499,21 +499,7 @@ public AccessTokenDO getLatestAccessToken(String consumerKey, AuthenticatedUser
 
             prepStmt = connection.prepareStatement(sql);
             prepStmt.setString(1, getPersistenceProcessor().getProcessedClientId(consumerKey));
-            String rootTenantDomain = (String) IdentityUtil.threadLocalProperties.get()
-                    .get(OrganizationManagementConstants.ROOT_TENANT_DOMAIN);
-            int appTenantId = IdentityTenantUtil.getLoginTenantId();
-            if (rootTenantDomain != null) {
-                appTenantId = OAuth2Util.getTenantId(rootTenantDomain);
-                if (log.isDebugEnabled()) {
-                    log.debug("Root tenant domain is not null. Therefore application ID is: " + appTenantId);
-                }
-            } else if (authzUser.isOrganizationUser()) {
```

Comment:
This logic added from this PR https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2413/files
Verified sub-org user deletion and token get revoked in the on-prem product. Need to verify in an IDaaS setup also. 
As I remember, that issue occured in IDaaS setup due to it was running on legacy-authz-runtime, but after new runtime, this fix is not required.

Reply:
We need to improve the queries to resolve the app resident tenant ID by joining the queires with idn-ouath-consumer-app table by app-id.